Sources/SwiftyXPC/XPCFileDescriptor.swift

```diff
@@ -7,10 +7,17 @@
 
 import Darwin
 
-/// A wrapper around a file descriptor, which can be embedded within XPC messages to send file descriptors to other processes.
+/// A owned wrapper around a file descriptor, which can be embedded within XPC messages to send file descriptors to other processes.
 ///
-/// Included for compatibility with macOS 10.15.
-/// On versions of macOS greater than 11.0, this class is unnecessary, and you can simply use a `FileDescriptor` from the `System` module instead.
+/// This wrapper takes ownership the file descriptor so user doesn't need to close the fd.
+///
+/// This wrapper is served for two purposes:
+/// 1. Provide compatibility with macOS 10.15.
+/// 2. Close the file descriptor when the message is embedded.
+///
+/// On versions of macOS greater than 11.0, you can also simply use a `FileDescriptor` from the `System` module. However, the `FileDescriptor`
+/// is not an owned descriptor so you have to make sure the file description is not closed in the sender process before the message is embedded
+/// in the XPC messages.
 public final class XPCFileDescriptor: Codable {
     /// The wrapped raw file descriptor.
     public let fileDescriptor: Int32
```

Comment:
If we are going to have this class auto-close its file descriptor on deallocation, we shouldn't keep its internal `fileDescriptor` property public, because if a client uses this, they will get a nasty surprise when the descriptor suddenly closes behind their back.

```suggestion
    internal let fileDescriptor: Int32
```